### DISCUSSION TITLE
Best way to include nlohmann/json into a different cmake project

### DISCUSSION THREAD

**Discussion by eike-fokken** [nlohmann/json#2670] [Q&A] [answered]
Hi there!
I'm building a numerical library that uses this wonderful json library for data input.
The only problem with nlohmann/json I have, is its huge .git size.

Up to now I include nlohmann/json as a submodule. This bumbs our .git folder from 16MB to 240MB.
Unfortunately cloning with the key `shallow = true` in `.gitmodules` halves the size of the git repo but doesn't actually reduce cloning time (which is 2 minutes or so instead of 10 seconds).

On the other hand, manually including the header file would work like a charm but I have to manually apply updates, which I will probably forget...

How do you (as in other users of nlohmann/json) handle this?

**Comment by nlohmann** [comment, marked as answer]
I think including just the headers is the best approach. Releases are sparse, so even if an update takes a few minutes, you will save this time quite soon.

**Reply by eike-fokken** [reply]
My problem with this is, that I will probably forget to update the included headers.
